File: ios/NHSOnline/NHSOnline/Support/UiInteraction/LoggedInHeaderStrategy.swift
import Foundation

class LoggedInHeaderStrategy: HeaderStrategyProtocol {
    var viewController: HomeViewController
    
    init(controller: HomeViewController) {
           self.viewController = controller
    }
    
    func apply(integrationLevel: IntegrationLevel) {
        switch integrationLevel {
        case .Gold,
             .SilverWithoutWebNavigation,
             .SilverWithWebNavigation:
            viewController.setVisibilityOfHeaderAndMenuBars(headerType: .Full)
        case .GoldOverlay:
            viewController.setVisibilityOfHeaderAndMenuBars(headerType: .SlimBack)
        case .Bronze, .GoldWithNoHeaders, .Unknown:
            viewController.setVisibilityOfHeaderAndMenuBars(headerType: .None)
        }
    }
}

File: android/app/src/main/java/com/nhs/online/nhsonline/support/uiinteraction/LoggedInHeaderStrategy.kt
package com.nhs.online.nhsonline.support.uiinteraction

import android.util.Log
import com.nhs.online.nhsonline.activities.HeaderIcon
import com.nhs.online.nhsonline.interfaces.IInteractor
import com.nhs.online.nhsonline.services.knownservices.enums.IntegrationLevel

private val TAG = LoggedInHeaderStrategy::class.java.simpleName

class LoggedInHeaderStrategy(private val uiInteractor: IInteractor): IHeaderStrategy {
    override fun apply(integrationLevel: IntegrationLevel) {
        when (integrationLevel) {
            IntegrationLevel.Gold, IntegrationLevel.SilverWithWebNavigation, IntegrationLevel.SilverWithoutWebNavigation -> {
                uiInteractor.showHeader()
                uiInteractor.showMenuBar()
            }
            IntegrationLevel.GoldOverlay -> uiInteractor.showHeaderSlim(HeaderIcon.Back)
            IntegrationLevel.GoldWithNoHeaders -> uiInteractor.hideHeaderAndMenu()
            else -> {
                Log.e(TAG, "Invalid Webview integration level: ${integrationLevel}")
                uiInteractor.hideHeaderAndMenu()
            }
        }
    }
}

